parquet-hadoop/src/main/java/org/apache/parquet/hadoop/util/SerializationUtil.java

```diff
@@ -97,7 +98,12 @@ public static void writeObjectToConfAsBase64(String key, Object obj, Configurati
       bais = new ByteArrayInputStream(bytes);
       gis = new GZIPInputStream(bais);
       ois = new ObjectInputStream(gis);
-      return (T) ois.readObject();
+      try {
+        return (T) ois.readObject();
+      } catch (ClassNotFoundException e) {
+        ClassLoaderObjectInputStream clois = new ClassLoaderObjectInputStream(conf.getClassLoader(), gis);
```

Comment:
I don't think the Configuration's ClassLoader is appropriate here either. The problem is that framework classes are loaded in some cases by a ClassLoader that is a parent to the ClassLoader used for application or user classes. Anything that is shipped in the user Jar could get ignored in that case. I think the right thing to do for now is to use the Thread context ClassLoader.

Also, this method should use whatever ClassLoader is passed in rather than trying the SerializationUtil's loader and then trying another one. All ClassLoader delegation should happen in the ClassLoader, not the calling class.

I recommend adding a ClassLoader option to this method and using that with the ClassLoaderObjectInputStream, and then adding a replacement method that defaults the ClassLoader to the Thread context's ClassLoader. Does that sound like it will work for your application?